src/main/resources/META-INF/rewrite/javaee-to-quarkus.yml

```diff
@@ -0,0 +1,181 @@
+      groupId: io.quarkus
+      artifactId: quarkus-hibernate-orm
+  - org.openrewrite.java.dependencies.AddDependency:
+      groupId: io.quarkus
+      artifactId: quarkus-jdbc-h2
+  - org.openrewrite.java.dependencies.AddDependency:
+      groupId: io.quarkus
+      artifactId: quarkus-junit5
+      scope: test
+  - org.openrewrite.java.dependencies.AddDependency:
+      groupId: io.rest-assured
+      artifactId: rest-assured
+      scope: test
+
+  # Add Maven Plugins
+  - org.openrewrite.maven.AddPlugin:
+      groupId: io.quarkus.platform
+      artifactId: quarkus-maven-plugin
+      version: '2.16.12.Final'
+      executions: <executions><execution><goals><goal>build</goal><goal>generate-code</goal><goal>generate-code-tests</goal></goals></execution></executions>
+  - org.openrewrite.maven.AddPlugin:
+      groupId: org.apache.maven.plugins
+      artifactId: maven-compiler-plugin
+      version: '3.10.1'
+      configuration: <configuration><compilerArgs><arg>-parameters</arg></compilerArgs></configuration>
+  - org.openrewrite.maven.AddPlugin:
+      groupId: org.apache.maven.plugins
+      artifactId: maven-surefire-plugin
+      version: '3.0.0-M7'
+      configuration: <configuration><systemPropertyVariables><java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager><maven.home>${maven.home}</maven.home></systemPropertyVariables></configuration>
+  - org.openrewrite.maven.AddPlugin:
+      groupId: org.apache.maven.plugins
+      artifactId: maven-failsafe-plugin
+      version: '3.0.0-M7'
+      executions: <executions><execution><goals><goal>integration-test</goal><goal>verify</goal></goals></execution></executions>
+      configuration: <configuration><systemPropertyVariables><native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path><java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager><maven.home>${maven.home}</maven.home></systemPropertyVariables></configuration>
+  - org.openrewrite.maven.AddProfile:
+      id: native
+      activation: <activation><property><name>native</name></property></activation>
+      properties: <properties><skipITs>false</skipITs><quarkus.package.type>native</quarkus.package.type></properties>
+  - org.openrewrite.quarkus.ConfigureQuarkusMavenPluginWithReasonableDefaults
+  - org.openrewrite.maven.BestPractices
+
+  # Remove JavaEE dependencies
+  - org.openrewrite.java.dependencies.RemoveDependency:
+      groupId: javax*
+      artifactId: javaee-api
+  - org.openrewrite.java.dependencies.RemoveDependency:
+      groupId: javax*
+      artifactId: cdi-api
+  - org.openrewrite.java.dependencies.RemoveDependency:
+      groupId: javax*
+      artifactId: javax*
+
+  # Prep for Java 11 upgrade
+  - org.openrewrite.maven.AddProperty:
+      key: maven.compiler.source
+      value: 11
+  - org.openrewrite.maven.AddProperty:
+      key: maven.compiler.target
+      value: 11
```

Comment:
Some of the changes above should already be part of `org.openrewrite.java.migrate.Java8toJava11` which is also executed below; can we reduce the duplication between those two?

Reply:
And note that if we need to break up `Java8toJava11` into better reusable parts for that to be used here, and perhaps in quarkus-updates, then that's certainly an option as well. I think it would help everyone if we reuse as much as possible, in large chunks, where we can.

Reply:
I added `maven.compiler.source` and `maven.compiler.target` properties because as per [Maven Compiler Plugin Release Configuration](https://docs.openrewrite.org/recipes/java/migrate/maven/usemavencompilerpluginreleaseconfiguration), these values only get updated if they exist. One of my tests was failing because these properties did not exist in the POM.